datasafe-simple-adapter/datasafe-simple-adapter-impl/src/main/java/de/adorsys/datasafe/simple/adapter/impl/SimpleDatasafeServiceImpl.java

```diff
@@ -50,8 +49,7 @@
     private static final ReadStorePassword universalReadStorePassword = new ReadStorePassword("secret");
     private static final String S3_PREFIX = "s3://";
 
-    private URI systemRoot;
-    private StorageService storageService;
+    private AbstractMap.Entry<URI, StorageService> rootAndStorage;
```

Comment:
Maybe it is better to create private static class/POJO for this instead of using tuple-like structure. From class field names it can be easily understood what URI is as e.g.

Reply:
Ok, I thought its less code. But of course, makes sense.